src/main/java/org/jenkinsci/plugins/github/config/GitHubServerConfig.java

```diff
@@ -257,6 +257,7 @@ public FormValidation doVerifyCredentials(
                 @QueryParameter String apiUrl, @QueryParameter String credentialsId) throws IOException {
 
             GitHubServerConfig config = new GitHubServerConfig(credentialsId);
+            config.setCustomApiUrl(isUrlCustom(apiUrl));
```

Comment:
when this component will be in jenkins, it will be much easier to avoid such logic here